### DISCUSSION TITLE
Automatically enable/disable dark mode via prefers-color-scheme

### DISCUSSION THREAD

**Discussion by rugk** [forem/forem#15318] [Feature Requests]
**Is your feature request related to a problem? Please describe.**

There is a manual switch for a dark mode, but that is inconvenient and I would have to do that on each site if each site were doing it like this.

**Describe the solution you'd like**

There is a quite new CSS media query feature called [`prefers-color-scheme`](https://developer.mozilla.org/docs/Web/CSS/@media/prefers-color-scheme). It allows you to detect whether the system/user wants a dark website, so you can adjust your CSS.

Currently it is only supported by Safari and Firefox 67 or higher.

Anyway, it would be great, if you could make use of this to automatically select the correct style.

**Describe alternatives you've considered**

Keep manual switch -> ugly. (Note: You can still keep it, so the user can overwrite it, but you can change the default.)

**Additional context**

BTW: I also have [a Firefox add-on called “Dark Mode Website Switcher”](https://addons.mozilla.org/de/firefox/addon/dark-mode-website-switcher/?src=external-github-issue) ([source code](https://github.com/rugk/website-dark-mode-switcher))  that you can use to toggle this setting in Firefox directly – without needing to change the system option. Maybe that could be helpful.

**Comment by sunchayn** [comment]
It's very practical and useful. However, I don't think it's sane to delivery production-level code based on experimental features. They are not stable and may experience behavioral changes in the future.

**Comment by hippotastic** [comment]
Just a heads up: This feature is no longer experimental. It's fully supported on all relevant browsers both on mobile and desktop, with zero problems reported about this CSS feature.

I think it's time to finally add this to dev.to, so I'm trying to bring some life back into this issue with my comment. :-)

**Comment by WA9ACE** [comment]
I agree. With places like [duckduckgo.com supporting it](https://spreadprivacy.com/duckduckgo-search-improvements/), and bugzilla.mozilla.org, and MDN showing full support in Firefox, Chromium browsers, and Safari I agree this feature should be given serious consideration.

**Comment by hippotastic** [comment]
Sorry to comment again, but what is the process to bring this to the attention of someone who can make the change happen? I find it a bit ironic that dev.to displays the „sore eyes?“ dark mode advertisement to me, while completely ignoring my preferences for dark mode already sent by my browser.

**Comment by WA9ACE** [comment]
> What is the process to bring this to the attention of someone who can make the change happen?

I gotchu fam 👍

**Comment by xgqfrms** [comment]
# switchable  css dark theme in js

> demo

https://codepen.io/xgqfrms/pen/rNxWzzg

**Comment by vasco3** [comment]
It's annoying to have to set it from the settings everytime the light in my room changes
 
It could be a keyboard shortcut to toggle between dark and light theme. Or quicker access to theme selection (less clicks) and without having to reload the page.
   
The Dark-mode toggle is a great solution.

**Comment by joshpuetz** [comment]
FWIW I'd LOVE to see this feature! I'm using a browser plugin called Dark Reader (https://darkreader.org), but having the default forem color scheme auto switch light to dark would be really nice (and handy on platforms that don't support plugins like iOS).

**Comment by marcelo-earth** [comment]
I would love to see this on Forem!

An option to get the system theme, today it is much more supported than when this post was published. I think it would save a lot, especially for those of us who spend changing light/dark mode at night or in environments with little darkness.
